From 4ca89a9da421f75e5486db49fd82b7c7ab37fb75 Mon Sep 17 00:00:00 2001
From: Soren Hansen <soren@linux2go.dk>
Date: Thu, 12 Aug 2010 11:00:49 +0200
Subject: [PATCH] Remove wrong check for uml monitor response size

The current check for the size of the response from the uml monitor is
problematic for a couple of reasons:

First of all, for me, the call to recvfrom returns 0, even though the
buffer actually is populated with the response from the monitor.

Second, it seems to me that (assuming recvfrom actually returned the
number of bytes read) it should be compared against the size of the
response header plus the actual data length rather than the max size of
the datagram.

At any rate, the only way I can get anything useful to happen here is to
completely remove the check, so this patch does just that.

Signed-off-by: Soren Hansen <soren@linux2go.dk>
---
 src/uml/uml_driver.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

Index: libvirt-0.8.3/src/uml/uml_driver.c
===================================================================
--- libvirt-0.8.3.orig/src/uml/uml_driver.c	2010-07-25 10:59:46.000000000 +0200
+++ libvirt-0.8.3/src/uml/uml_driver.c	2010-08-16 13:33:08.696132001 +0200
@@ -737,10 +737,6 @@
             virReportSystemError(errno, _("cannot read reply %s"), cmd);
             goto error;
         }
-        if (nbytes < sizeof res) {
-            virReportSystemError(0, _("incomplete reply %s"), cmd);
-            goto error;
-        }
         if (sizeof res.data < res.length) {
             virReportSystemError(0, _("invalid length in reply %s"), cmd);
             goto error;
